app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarViewModel.kt

```diff
@@ -168,23 +210,42 @@ sealed class VaultUnlockedNavBarAction {
  * Models events for the bottom tab of the vault unlocked portion of the app.
  */
 sealed class VaultUnlockedNavBarEvent {
+    /**
+     * Whether to return to the root of the subgraph or not.
+     */
+    abstract val returnToSubgraphRoot: Boolean
     /**
      * Navigate to the Generator screen.
      */
-    data object NavigateToGeneratorScreen : VaultUnlockedNavBarEvent()
+    data class NavigateToGeneratorScreen(override val returnToSubgraphRoot: Boolean) :
+        VaultUnlockedNavBarEvent()
 
     /**
      * Navigate to the Send screen.
      */
-    data object NavigateToSendScreen : VaultUnlockedNavBarEvent()
+    data class NavigateToSendScreen(override val returnToSubgraphRoot: Boolean) :
+        VaultUnlockedNavBarEvent()
 
     /**
      * Navigate to the Vault screen.
      */
-    data object NavigateToVaultScreen : VaultUnlockedNavBarEvent()
+    data class NavigateToVaultScreen(override val returnToSubgraphRoot: Boolean) :
+        VaultUnlockedNavBarEvent()
 
     /**
      * Navigate to the Settings screen.
      */
-    data object NavigateToSettingsScreen : VaultUnlockedNavBarEvent()
+    data class NavigateToSettingsScreen(override val returnToSubgraphRoot: Boolean) :
+        VaultUnlockedNavBarEvent()
+}
+
+/**
+ * Enum to represent the current bottom tab that is selected.
+ */
+@Parcelize
+enum class BottomNavDestination : Parcelable {
+    VAULT,
+    SEND,
+    GENERATOR,
+    SETTINGS,
```

Comment:
Do we still need this? I thought this would all be covered by just the boolean being passed around.